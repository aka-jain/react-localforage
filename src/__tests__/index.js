/* eslint-env jest */
// eslint-disable-next-line no-unused-vars
import React from 'react'
import Forage from '../index'
import renderer from 'react-test-renderer'
const TestUtils = require('react-dom/test-utils')

describe('Forage.GetItem', () => {
  test('renders', () => {
    const tree = renderer.create(<Forage.GetItem lazy />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with children', () => {
    const renderCallback = result => {
      return (
        <pre>
          {JSON.stringify(result)}
        </pre>
      )
    }
    const tree = renderer
      .create(
        <Forage.GetItem lazy>
          {renderCallback}
        </Forage.GetItem>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can render with no children', () => {
    TestUtils.renderIntoDocument(<Forage.GetItem lazy />)
  })
})

describe('Forage.SetItem', () => {
  test('renders', () => {
    const tree = renderer.create(<Forage.SetItem lazy/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with children', () => {
    const renderCallback = result => {
      return (
        <pre>
          {JSON.stringify(result)}
        </pre>
      )
    }
    const tree = renderer
      .create(
        <Forage.SetItem lazy>
          {renderCallback}
        </Forage.SetItem>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can render with no children', () => {
    TestUtils.renderIntoDocument(<Forage.SetItem lazy/>)
  })
})
