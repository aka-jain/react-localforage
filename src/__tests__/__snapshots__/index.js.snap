// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forage.GetItem renders 1`] = `null`;

exports[`Forage.GetItem renders with children 1`] = `null`;

exports[`Forage.SetItem renders 1`] = `null`;

exports[`Forage.SetItem renders with children 1`] = `null`;

exports[`react-localforage renders 1`] = `null`;

exports[`react-localforage renders with children 1`] = `null`;
